From f09c81577d2ad50adf8d7517384386cdc926eb36 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Fri, 17 Jun 2022 05:43:26 +0300
Subject: [PATCH] [SW-0] rom: enable rr config for slave

Range register configuration in a security disabled system would
open up access to the entire LBW and HBW space . This is done,
as in gaudi3 by default the entire LBW and HBW is blocked by HW.

Currently the opening up of access is only done for master. This
patch implements the same for slave.

As this is only done for security disabled systems, it does not
have any impact on a production system.

Change-Id: I2c89f33bdd1cb6111b0a743472079bb05ea37c68
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index dc82639..bf2ff49 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -164,11 +164,10 @@
 	int sec_en = security_efuse_security_en();
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_IN_ROM_RECOVERY);
 
-	if ((!sec_en) && (!arcboot_rom_global.die_id))
+	if (!sec_en) {
 		security_rr_recovery_config();
-
-	if (!sec_en)
 		rom_pcie_recovery();
+	}
 
 	rom_d2d_link_init();
 
